### PR TITLE
cclient: Implement iota_api_get_transactions_to_approve

### PR DESCRIPTION
**Pull request by jkrvivian**
Implement iota_api_get_transactions_to_approve

### CODE REVIEW

File: cclient/request/get_transactions_to_approve.c [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ */
+
+#include "get_transactions_to_approve.h"
+
+get_transactions_to_approve_req_t* get_transactions_to_approve_req_new() {
+  get_transactions_to_approve_req_t* req =
+      (get_transactions_to_approve_req_t*)malloc(
+          sizeof(get_transactions_to_approve_req_t));
+  if (req == NULL) {
+      return NULL;
```
Comment (oopsmonk) [inline cclient/request/get_transactions_to_approve.c]: Do you set up the commit hook?

Comment (jkrvivian) [reply cclient/request/get_transactions_to_approve.c]: My commit hook will stopped by `cpplint`([issue  213](https://github.com/iotaledger/entangled/issues/213)), and I haven't solve this problem yet.
So I checked it with running `clang-format` in command line and I missed this one.   
Sorry!

File: cclient/serialization/tests/test_json.c [hunk 2]
```diff
@@ -465,11 +466,58 @@ void test_deserialize_get_balances(void) {
 }
 
 void test_serialize_get_transactions_to_approve(void) {
-  // TODO
+  serializer_t serializer;
+  init_json_serializer(&serializer);
+  const char* json_text =
+      "{\"command\":\"getTransactionsToApprove\",\"depth\":15"
+      ",\"reference\":"
+      "\"TKGDZ9GEI9CPNQGHEATIISAKYPPPSXVCXBSR9EIWCTHHSSEQCD9YLDPEXYERCNJVASRGWM"
```
Comment (tsvisabo) [inline cclient/serialization/tests/test_json.c]: can you please define all those strings outside for a global cclient use?

File: cclient/request/get_transactions_to_approve.c [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  get_transactions_to_approve_req_t* req =
+      (get_transactions_to_approve_req_t*)malloc(
+          sizeof(get_transactions_to_approve_req_t));
+  if (req == NULL) {
+    return NULL;
+  }
+
+  req->reference = char_buffer_new();
+  if (req->reference == NULL) {
+    return NULL;
```
Comment (thibault-martinez) [inline cclient/request/get_transactions_to_approve.c]: You are leaking `req` here

File: cclient/request/get_transactions_to_approve.h [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+#define CCLIENT_REQUEST_GET_TRANSACTIONS_TO_APPROVE_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "types/types.h"
+
+typedef struct {
+  int32_t depth;
```
Comment (thibault-martinez) [inline cclient/request/get_transactions_to_approve.h]: Maybe unsign this ?

File: cclient/response/get_transactions_to_approve.c [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include "get_transactions_to_approve.h"
+
+get_transactions_to_approve_res_t* get_transactions_to_approve_res_new() {
+  get_transactions_to_approve_res_t* res =
+      (get_transactions_to_approve_res_t*)malloc(
+          sizeof(get_transactions_to_approve_res_t));
```
Comment (thibault-martinez) [inline cclient/response/get_transactions_to_approve.c:13]: Check malloc return

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: cclient/request/get_transactions_to_approve.h (hunk 4 above)
Comment (jkrvivian) [reply cclient/request/get_transactions_to_approve.h]: yeah! sure!

**Review by tsvisabo — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**